k8s-scheduler/src/test/java/com/vmware/dcm/SchedulerTest.java

```diff
@@ -1245,6 +1245,87 @@ public void testSchedulerDebugDump() {
         System.out.println(results);
     }
 
+    /*
+     * Make sure that autoscope gets topK for each node resource
+     */
+    @CartesianProductTest(name = "{0}")
+    public void testAutoScopeTopK(final TestArguments args) {
+
+        final IConnectionPool dbConnectionPool = setupDDlog();
+
+        final List<Integer> nodeCpuCapacities = (List<Integer>) args.get(1);
+        final List<Integer> nodeMemoryCapacities = (List<Integer>) args.get(2);
+        final List<Integer> nodeStorageCapacities = (List<Integer>) args.get(3);
+        final List<Integer> nodepodsCapacities = (List<Integer>) args.get(4);
+        final Iterator<Integer> cpuCapIt = nodeCpuCapacities.iterator();
+        final Iterator<Integer> memCapIt = nodeMemoryCapacities.iterator();
+        final Iterator<Integer> storageCapIt = nodeStorageCapacities.iterator();
+        final Iterator<Integer> podsCapIt = nodepodsCapacities.iterator();
+        final HashMap<String, List<Integer>> expected = new HashMap<>();
+        expected.put("cpu", (List<Integer>) args.get(5));
+        expected.put("memory", (List<Integer>) args.get(6));
+        expected.put("ephemeral-storage", (List<Integer>) args.get(7));
+        expected.put("pods", (List<Integer>) args.get(8));
+
+        final int limit = 3;
```

Comment:
Not sure, but is this limit large enough for the test (that is, does it only pull a subset of nodes out)?

Reply:
Yep it's large enough. The test case has 10 nodes.